eo-maven-plugin/src/main/java/org/eolang/maven/DcsDepgraph.java

```diff
@@ -0,0 +1,218 @@
+ * You can read more <a href="https://github.com/ferstl/depgraph-maven-plugin">here</a>
+ *
+ * @since 0.28.11
+ */
+final class DcsDepgraph implements Dependencies {
+
+    /**
+     * Maven project.
+     */
+    private final MavenProject project;
+
+    /**
+     * Maven session.
+     */
+    private final MavenSession session;
+
+    /**
+     * Maven plugin manager.
+     */
+    private final BuildPluginManager manager;
+
+    /**
+     * Directory to save all transitive dependencies files.
+     */
+    private final Path dir;
+
+    /**
+     * Dependency for which we are looking transitive dependencies.
+     */
+    private final Dependency dependency;
+
+    /**
+     * The main contructor.
+     *
+     * @param project Maven project
+     * @param session Maven session
+     * @param manager Maven plugin manager
+     * @param dir Directory to save all transitive dependencies files
+     * @param dependency Dependency
+     * @checkstyle ParameterNumberCheck (10 lines)
+     */
+    DcsDepgraph(
+        final MavenProject project,
+        final MavenSession session,
+        final BuildPluginManager manager,
+        final Path dir,
+        final Dependency dependency
+    ) {
+        this.project = project;
+        this.session = session;
+        this.manager = manager;
+        this.dir = dir;
+        this.dependency = dependency;
+    }
+
+    @Override
+    public Collection<Dependency> all() {
+        return new DcsJson(this.file(this.dependency)).all();
+    }
+
+    /**
+     * Receive file with dependencies jar file.
+     *
+     * @param origin Dependency
+     * @return Path to the saved json dependency file
+     */
+    private Path file(final Dependency origin) {
+        try {
+            final String name = DcsDepgraph.fileName(origin);
+            MojoExecutor.executeMojo(
+                MojoExecutor.plugin(
+                    MojoExecutor.groupId("com.github.ferstl"),
+                    MojoExecutor.artifactId("depgraph-maven-plugin"),
+                    MojoExecutor.version("4.0.2")
+                ),
+                MojoExecutor.goal("for-artifact"),
+                MojoExecutor.configuration(
+                    MojoExecutor.element("groupId", origin.getGroupId()),
+                    MojoExecutor.element("artifactId", origin.getArtifactId()),
+                    MojoExecutor.element("version", origin.getVersion()),
+                    MojoExecutor.element("graphFormat", "json"),
+                    MojoExecutor.element("outputDirectory", this.dir.toString()),
+                    MojoExecutor.element("outputFileName", name)
+                ),
+                MojoExecutor.executionEnvironment(
+                    this.project,
+                    this.session,
+                    this.manager
+                )
+            );
+            return this.dir.resolve(name);
+        } catch (final MojoExecutionException ex) {
+            throw new IllegalStateException(ex);
```

Comment:
@volodya-lombrozo Can we specify this error message according to this: https://www.yegor256.com/2015/12/01/rethrow-exceptions.html

Reply:
@Graur Done